bitcoin/src/blockdata/transaction/extended.rs

```diff
@@ -0,0 +1,133 @@
+//! Extended represenatation of transaction.
+//!
+//! The usual Bitcoin transaction lack data that is required in many scenarios. The most famous is
+//! computing transaction fee which is imposible without additional context.
+//!
+//! To help with these cases this module provides `ExtendedTxIn` and `ExtendedTransaction` which
+//! contain extra data.
+
+use super::{Version, TxIn, TxOut, Txid, Transaction, InputWeightPrediction};
+use crate::{absolute, FeeRate, Weight, VarInt, Amount};
+use crate::amount::CheckedSum;
+
+/// Transaction input extended with additional data.
+///
+/// The type parameter allows adding user-defined data which may be useful for signing.
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct ExtendedTxIn<ExtraData = ()> {
+    /// The non-extended transaction input.
+    pub txin: TxIn,
+
+    /// The output this input is spending.
+    pub previous_output: TxOut,
+
+    /// User-selected data such as key derivation path.
+    pub extra: ExtraData,
```

Comment:
maybe we could also add  `max_satisfaction_weight`?

Reply:
No, it's not generally known. But you can make your own `struct UtxoInfo { kind: AddressKind, path: DerivationPath }` to facilitate both coin selection and signing. I guess I will make some examples of how to use these things.